File: app/src/main/java/com/example/pma_project/HistoryActivity.java
package com.example.pma_project;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

import static com.example.pma_project.MainActivity.data;

public class HistoryActivity extends AppCompatActivity {

    private RecyclerView dataEntries;
    private RecyclerView.Adapter adapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_history);

        ArrayList<Data> dataEntries = data;

        this.dataEntries = (RecyclerView)findViewById(R.id.dataEntries);
        RecyclerView.LayoutManager mLayoutManager = new LinearLayoutManager(this);
        this.dataEntries.setLayoutManager(mLayoutManager);

        adapter = new DataAdapter(dataEntries);
        this.dataEntries.setAdapter(adapter);
    }

}

File: app/src/main/java/com/example/pma_project/HistoryActivity.kt
package com.example.pma_project

import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.RecyclerView
import android.os.Bundle
import android.view.View
import com.example.pma_project.R
import com.example.pma_project.MainActivity
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.pma_project.DataAdapter

class HistoryActivity : AppCompatActivity() {
    private var dataEntries: RecyclerView? = null
    private var adapter: RecyclerView.Adapter<*>? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_history)
        val dataEntries = MainActivity.data
        this.dataEntries = findViewById<View>(R.id.dataEntries) as RecyclerView
        val mLayoutManager: RecyclerView.LayoutManager = LinearLayoutManager(this)
        this.dataEntries!!.layoutManager = mLayoutManager
        adapter = DataAdapter(dataEntries)
        this.dataEntries!!.adapter = adapter
    }
}